src/krux/settings.py

```diff
@@ -210,17 +216,15 @@ def delete(self, namespace, setting_name):
                 self.dirty = True
 
     def update_file_location(self, location):
-        """Assumes settings.persist.location will be changed to location:
-        tries to delete current persistent settings file
-        then updates file_location attribute
-        """
-        if "/" + location + "/" != self.file_location:
-            try:
-                if os.stat(self.file_location + SETTINGS_FILENAME):
-                    os.remove(self.file_location + SETTINGS_FILENAME)
-            except:
-                pass
-            self.file_location = "/" + location + "/"
+        """Delete any unused persistent settings when set location"""
+        self.file_location = Store.get_vfs_location(location)
+        try:
+            if FLASH_PATH in self.file_location:
```

Comment:
Why the data transformations then search withing a string when you can compare directly?
`if location == FLASH_PATH:`

Reply:
Nevermind! Noted you're also updating the "file_location" variable